modules/tinymce/src/plugins/fullscreen/test/ts/browser/FullScreenPluginTest.ts

```diff
@@ -76,35 +132,68 @@ UnitTest.asynctest('browser.tinymce.plugins.fullscreen.FullScreenPluginTest', (s
         lastEventArgs.set(e);
       });
     }
-  });
+  };
 
-  Pipeline.async({}, [
-    Log.chainsAsStep('TBA', 'FullScreen: Toggle fullscreen on, open link dialog, insert link, close dialog and toggle fullscreen off', [
-      cSetupEditor,
-      Chain.fromParent(Chain.identity, [
-        cAssertFullscreenClass('Before fullscreen command', false),
-        Chain.op((editor: Editor) => editor.execCommand('mceFullScreen', true)),
-        cAssertEditorAndLastEvent('After fullscreen command', true),
-        cAssertFullscreenClass('After fullscreen command', true),
-        Chain.op((editor: Editor) => editor.execCommand('mceLink', true)),
-        cWaitForDialog('Insert/Edit Link'),
-        cCloseOnlyWindow,
-        cAssertFullscreenClass('After window is closed', true),
-        Chain.op((editor: Editor) => editor.execCommand('mceFullScreen')),
-        cAssertEditorAndLastEvent('After fullscreen toggled', false),
-        cAssertFullscreenClass('After fullscreen toggled', false)
-      ]),
+  const standardConfig: Config = {
+    label: 'Standard',
+    setupEditor: () => McEditor.cFromSettings(settings),
+    cleanupEditor: () => McEditor.cRemove
+  };
+
+  const shadowRootConfig: Config = {
+    label: 'ShadowHost',
+    setupEditor: () => {
+      const shadowHost = SugarElement.fromTag('div');
+      Insert.append(SugarBody.body(), shadowHost);
+      const sr = SugarElement.fromDom(shadowHost.dom.attachShadow({ mode: 'open' }));
+      const editorDiv = SugarElement.fromTag('div');
+      Insert.append(sr, editorDiv);
+      return McEditor.cFromElement(editorDiv, settings);
+    },
+    cleanupEditor: () => Chain.fromChains([
+      Chain.op((editor: Editor) => {
```

Comment:
This is correct? I think this needs to happen in the opposite order, otherwise the editor will be removed from the DOM before it's shutdown properly.

Reply:
Thanks for picking this up. It should be fixed now.